File: HomeWorks/CPP/Division without Div-operator.cpp
#include<iostream>
using namespace std;
int main()
{
    int num1, num2, ans=0;
    cout<<"pls enter number1 : ";
    cin>>num1;
    cout<<"                   --- ="<<endl;
    cout<<"pls enter number2 : ";
    cin>>num2;
    if(num2 == 0)
    {
        cout<<endl<<"D-Bug : num2 == 0"<<endl;
        cout<<"answer is undefined."<<endl;
    }

    else if(num1 == 0)
    {
        cout<<endl<<"D-Bug : num1 == 0"<<endl;
        cout<<"Answer = 0"<<endl;
    }

    else if(num1 == num2)
    {
        cout<<endl<<"D-Bug : num1 == num2"<<endl;
        cout<<"Answer = 1"<<endl;
    }

    else if(num2 == 1)
    {
        cout<<endl<<"D-Bug : num2 == 1"<<endl;
        cout<<"Answer = "<<num1<<endl;
    }

    else if(num1>num2)
    {
        cout<<endl<<"D-Bug : num1 > num2"<<endl;
        while(num1>=0)
        {
            num1-=num2;
            ans++;
        }
        if((num1+num2)==0) // یعنی باقی مانده 0 شده
        {
            cout<<"Answer = "<<ans-1<<endl<<"Remaining = "<<num1+num2;;
        }
        else
        {
            cout<<"Answer ≈ "<<ans-1<<endl<<"Remaining = "<<num1+num2;
            /*                                   |
										  +--> باقی مانده  */
        }

    }

    else if(num2>num1)
    {
        cout<<endl<<"D-Bug : num1 < num2"<<endl;
        cout<<"Answer < 1   =) "<<endl;
    }
	
	else
	{
		cout<<"wtf are u doing??";
	}
}
//MadMad_67

File: homeworks/py/division_without _div_operator.py
num1 = int(input("pls enter number1 : "))
print("                   ---- =")
num2 = int(input("pls enter number2 : "))
ans = 0

if(num2 == 0):
    print("answer is undefined.")
elif(num1 == 0):
    print("Answer = 0")
elif(num1==num2):
    print("Answer = 1")
elif(num2==1):
    print("Answer = ", num1)
elif(num1>num2):
    while(num1>=0):
        num1-=num2
        ans+=1
    if((num1+num2)==0):
        print("Answer = ",ans-1, "\nRemaining = ", num1+num2)
    else:
        print("Answer ≈ ",ans-1, "\nRemaining = ", num1+num2)
elif(num2>num1):
    print("Answer < 1")
else:
    print("wtf are u doing??")
#MadMad_26